components/layout/query.rs

```diff
@@ -927,44 +931,48 @@ pub fn process_element_inner_text_query<N: LayoutNode>(node: N,
     inner_text.into_iter().collect()
 }
 
-// https://html.spec.whatwg.org/multipage/#inner-text-collection-steps
 #[allow(unsafe_code)]
-fn inner_text_collection_steps<N: LayoutNode>(node: N,
-                                              indexable_text: &IndexableText,
-                                              results: &mut Vec<InnerTextItem>) {
-    let mut items = Vec::new();
-    for child in node.traverse_preorder() {
-        let node = match child.type_id() {
-            LayoutNodeType::Text => {
-                child.parent_node().unwrap()
-            },
-            _ => child,
-        };
+fn get_style<N: LayoutNode>(node: &N) -> Option<ServoArc<ComputedValues>> {
+    let node = match node.type_id() {
+        LayoutNodeType::Text => {
+            node.parent_node().unwrap()
+        },
+        _ => *node,
+    };
 
-        let element_data = unsafe {
-            node.get_style_and_layout_data().map(|d| {
-                &(*(d.ptr.as_ptr() as *mut StyleData)).element_data
-            })
-        };
+    let element_data = unsafe {
```

Comment:
There's no need for unsafe if you use `as_element`. Afterwards you can use:

```rust
element.borrow_data()?.styles.primary().clone()
```

(if there's data, there's guaranteed styles, unless you're in the middle of style resolution, which you're not).